src/Sylius/Behat/Context/Api/Shop/CartContext.php

```diff
@@ -173,15 +176,27 @@ public function myCartShouldBeCleared(): void
 
     /**
      * @Then /^my (cart)'s total should be ("[^"]+")$/
+     * @Then /^my (cart) total should be ("[^"]+")$/
      */
     public function myCartSTotalShouldBe(string $tokenValue, int $total): void
     {
-        $response = $this->cartsClient->show($tokenValue);
+        $this->response = $this->cartsClient->show($tokenValue);
```

Comment:
IMO this change is unnecessary, you should leave here previous code, and in function `myDiscountShouldBe` use `cartsClient->getLastResponse` or get cart token from `sharedStorage` and make show call again.